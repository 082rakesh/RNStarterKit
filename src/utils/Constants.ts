export const IMG_BASE_URL =
  'https://media-assets.swiggy.com/swiggy/image/upload/fl_lossy,f_auto,q_auto,w_660/';
export const ALL_RESTURANTS =
  'https://www.swiggy.com/dapi/restaurants/list/v5?lat=12.96340&lng=77.58550&is-seo-homepage-enabled=true&page_type=DESKTOP_WEB_LISTING';
export const RESTUARANTS_DETAILS =
  'https://www.swiggy.com/dapi/menu/pl?page-type=REGULAR_MENU&complete-menu=true&lat=12.96340&lng=77.58550&restaurantId=';
export const SEARCH_RESTRAUNTS_PLACEHOLDER_TEXT = 'Search restraunts';
export const CATAGORY =
  'type.googleapis.com/swiggy.presentation.food.v2.ItemCategory';

export const BASE_URL = 'https://www.swiggy.com/dapi/';
export const RES_LIST =
  'restaurants/list/v5?lat=12.96340&lng=77.58550&is-seo-homepage-enabled=true&page_type=DESKTOP_WEB_LISTING';
export const DARK_THEME_TYPE = 'dark';

export const ALL_POSTS = 'https://jsonplaceholder.typicode.com/posts';

export enum ButtonType {
  PRIMARY = 'primary',
  SECONDARY = 'secondary',
  LARGE_PRIMARY = 'largePrimary',
  LARGE_SECONDARY = 'largeSecondary',
  LINK = 'link',
}
