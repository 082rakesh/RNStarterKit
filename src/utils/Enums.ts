export const enum APIStatus {
  LOADING = 'loading',
  SUCCESS = 'succeeded',
  REJECTED = 'failed',
}

export const enum TextTypes {
  PRIMARY = 'primary',
  SECONDARY = 'secondary',
  HEADING = 'heading',
}
