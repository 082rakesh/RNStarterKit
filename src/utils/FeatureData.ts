export interface Feature {
	name: string;
	id: number;
}

export interface Features {
	features: Feature[];
}

export const featureList: Features = {
	features: [
		{ name: 'Buttons', id: 1 },
		{ name: 'List', id: 2 },
		{ name: 'Sample', id: 3 },
		{ name: 'Custom hooks', id: 4 },
		{ name: 'Custom Fonts sample', id: 5 },
	],
};
