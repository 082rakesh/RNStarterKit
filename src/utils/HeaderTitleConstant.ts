export const TITLE_RESTAURANT = 'Restaurants';
export const TITLE_RES_DETAILS = 'Restaurants Details';
export const TITLE_COMPONENTS = 'Components';
export const TITLE_COMP_DETAILS = 'Component Details';
