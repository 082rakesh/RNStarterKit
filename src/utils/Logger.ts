import {logger, fileAsyncTransport} from 'react-native-logs';
import FileSystem from 'react-native-fs';

const today = new Date();
const dateString = today.toISOString().split('T')[0];

const config = {
  severity: 'debug',
  transport: fileAsyncTransport,
  levels: {
    custom: 0,
    debug: 1,
    info: 2,
    warn: 3,
    error: 4,
    devNotice: 5,
  },
  transportOptions: {
    colors: {
      custom: 'white',
      devNotice: 'blue',
      info: 'blueBright',
      warn: 'yellowBright',
      error: 'redBright',
      debug: 'white',
    },
    FS: FileSystem,
    fileName: `logs_${dateString}.txt`, // Create a new file every day
  },
};

const log = logger.createLogger(config);
const filePath = FileSystem.DocumentDirectoryPath + '/logs.txt';
console.log('filePath', filePath);

export const Logger = {
  debug: (message: string) => {
    log.debug(message);
  },
  info: (message: string) => {
    log.info(message);
  },
  warn: (message: string) => {
    log.info(message);
  },
  error: (message: string) => {
    log.info(message);
  },
};
