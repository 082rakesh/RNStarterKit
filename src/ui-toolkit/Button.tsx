import {StyleSheet, Pressable, ViewStyle} from 'react-native';
import React from 'react';
import {ButtonStyle} from './styles/ButtonStyle';
import {ButtonType} from '../utils/Constants';

export type Props = {
  children: React.ReactElement;
  onPress?: () => void;
  style?: ViewStyle;
  type?: ButtonType;
};

const Button = ({
  children,
  onPress,
  style = {},
  type = ButtonType.PRIMARY,
}: Props) => {
  return (
    <Pressable style={[styles[type], {...style}]} onPress={onPress}>
      {children}
    </Pressable>
  );
};

const styles = StyleSheet.create({
  primary: {
    ...ButtonStyle.style.primary,
  },
  secondary: {
    ...ButtonStyle.style.secondary,
  },
  largePrimary: {
    ...ButtonStyle.style.largePrimary,
  },
  largeSecondary: {
    ...ButtonStyle.style.largeSecondary,
  },
  link: {
    ...ButtonStyle.style.link,
  },
});

const buttonStyle = StyleSheet.compose({
  
})
export default  Button;
