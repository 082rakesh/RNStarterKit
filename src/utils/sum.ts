import axios from 'axios';
import { ALL_POSTS } from '../utils/Constants';

export const getPosts = async () => {
	try {
		const response = axios.get(ALL_POSTS);
		const jsonResponse = (await response).data;
		return jsonResponse;
	} catch (error) {
		throw new Error('Failed to fetch post data');
	}
};

export const getPeople = async () => {
	try {
		const response = axios.get('https://swapi.dev/api/people/1');
		const jsonResponse = (await response).data;
		return jsonResponse.name;
	} catch (error) {
		throw new Error('Failed to fetch people name');
	}
};
