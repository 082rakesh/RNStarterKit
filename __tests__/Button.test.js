import React from 'react';
import renderer from 'react-test-renderer';
import Button from '../src/ui-toolkit/Button';
import { ButtonType } from '../src/utils/Constants';

test('render LARGE PRIMARY Button component', () => {
	const tree = renderer
		.create(<Button type={ButtonType.LARGE_PRIMARY} />)
		.toJSON();
	expect(tree).toMatchSnapshot();
});

test('render PRIMARY Button component', () => {
	const tree = renderer.create(<Button type={ButtonType.PRIMARY} />).toJSON();
	expect(tree).toMatchSnapshot();
});

test('render SECONDARY Button component', () => {
	const tree = renderer.create(<Button type={ButtonType.SECONDARY} />).toJSON();
	expect(tree).toMatchSnapshot();
});

test('render LARGE SECONDARY Button component', () => {
	const tree = renderer
		.create(<Button type={ButtonType.LARGE_SECONDARY} />)
		.toJSON();
	expect(tree).toMatchSnapshot();
});

test('render LINK Button component', () => {
	const tree = renderer.create(<Button type={ButtonType.LINK} />).toJSON();
	expect(tree).toMatchSnapshot();
});
