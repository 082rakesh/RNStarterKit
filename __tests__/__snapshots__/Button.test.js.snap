// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render LARGE PRIMARY Button component 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      {
        "alignItems": "center",
        "backgroundColor": "red",
        "borderRadius": 5,
        "padding": 10,
        "width": 200,
      },
      {},
    ]
  }
/>
`;

exports[`render LARGE SECONDARY Button component 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      {
        "alignItems": "center",
        "backgroundColor": "transparent",
        "borderRadius": 5,
        "borderWidth": 1,
        "padding": 10,
        "width": 200,
      },
      {},
    ]
  }
/>
`;

exports[`render LINK Button component 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      {
        "alignItems": "center",
        "backgroundColor": "transparent",
        "padding": 10,
      },
      {},
    ]
  }
/>
`;

exports[`render PRIMARY Button component 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      {
        "alignItems": "center",
        "backgroundColor": "red",
        "borderRadius": 5,
        "padding": 10,
      },
      {},
    ]
  }
/>
`;

exports[`render SECONDARY Button component 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      {
        "alignItems": "center",
        "backgroundColor": "transparent",
        "borderRadius": 5,
        "borderWidth": 1,
        "padding": 10,
      },
      {},
    ]
  }
/>
`;
